File: melhor-consulta-angular/src/app/models/usuario.ts
export interface Usuario{}
export interface UsuarioAcesso{
  login: string,
  email: string,
  senha: string
}

export interface UsuarioCadastro{
  nome: string,
  sobrenome: string,
  email: string,
  senha: string,
  sexo: string,
  dtNascimento: Date
}

File: melhor-consulta-spring/src/main/java/com/soaresdev/melhorconsultaspring/models/Usuario.java
package com.soaresdev.melhorconsultaspring.models;

import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Date;

@Entity
@Data
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nome;
    private String sobrenome;
    private String email;
    private String senha;
    private String sexo;
    private Date dtNascimento;
    private String perfil;
    private boolean situacao;
    private Date dtInicio;
}
